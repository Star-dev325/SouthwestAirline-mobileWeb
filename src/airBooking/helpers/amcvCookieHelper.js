// @flow
import _ from 'lodash';
import * as Cookie from 'src/shared/swa-persistence/cookie';

import { buildPathWithQuery } from 'src/shared/helpers/pathUtils';

const AMCV_COOKIE_NAME = 'AMCV_65D316D751E563EC0A490D4C%40AdobeOrg';
const AMCV_COOKIE_KEY_SEPARATOR = 'MID';
const AMCV_COOKIE_PIPE_SEPARATOR = '|';
const CHASE_URL_KEYWORD = 'creditcard';

export type ChaseAppendParams = {
  isChaseCombo?: boolean,
  pageId?: string
};

export const getMcvid = () => {
  const value = Cookie.getValue(AMCV_COOKIE_NAME);

  return _.chain(value).split(AMCV_COOKIE_KEY_SEPARATOR).nth(1).split(AMCV_COOKIE_PIPE_SEPARATOR).nth(1).value() || '';
};

// Deprecated - please use appendParamsIfChaseUrl instead
export const addMvcidToChaseUrl = (url: string, isChase: boolean) => {
  let updateUrl = url;

  if (!_.isEmpty(url) && (isChase || url.includes(CHASE_URL_KEYWORD))) {
    const mcvid = getMcvid();

    if (!_.isEmpty(mcvid)) {
      updateUrl = `${url}&mcvid=${mcvid}`;
    }
  }

  return updateUrl;
};

export const appendParamsIfChaseUrl = (target: string = '', params: ChaseAppendParams = {}) => {
  const { isChaseCombo = false, pageId = '' } = params;
  const query = {};
  const isChaseUrl = _.includes(target, CHASE_URL_KEYWORD) || isChaseCombo;
  const mcvid = isChaseUrl && getMcvid();

  mcvid && _.set(query, 'mcvid', mcvid);
  pageId && _.set(query, 'pageId', pageId);

  return buildPathWithQuery(target, query);
};
