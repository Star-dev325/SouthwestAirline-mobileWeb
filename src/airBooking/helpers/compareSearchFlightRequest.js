import _ from 'lodash';
import { transformToAPIRequest } from 'src/shared/transformers/flightProductSearchRequestTransformer';

export const compareSearchFlightRequest = (requestOne, requestTwo) => {
  const transformedRequestOneQuery = _.get(transformToAPIRequest(requestOne), 'query');
  const transformedRequestTwoQuery = _.get(transformToAPIRequest(requestTwo), 'query');
  const requestOneWithoutPromoCode = _.omit(transformedRequestOneQuery, 'promo-code');
  const requestTwoWithoutPromoCode = _.omit(transformedRequestTwoQuery, 'promo-code');

  return _.isEqual(requestOneWithoutPromoCode, requestTwoWithoutPromoCode);
};
