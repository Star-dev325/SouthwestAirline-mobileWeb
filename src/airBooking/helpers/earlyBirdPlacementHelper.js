// @flow

import _ from 'lodash';
import numeral from 'numeral';
import i18n from '@swa-ui/locale';
import type { EarlyBirdEligibility } from 'src/airBooking/flow-typed/airBooking.types';
import type { CurrencyType } from 'src/shared/flow-typed/shared.types';

const getEarlyBirdPrefixText = (unitPriceInBound: CurrencyType, unitPriceOutBound: CurrencyType) => {
  const showStartingFromMessage =
    !!unitPriceOutBound &&
    !!unitPriceInBound &&
    _.get(unitPriceOutBound, 'amount') !== _.get(unitPriceInBound, 'amount');

  return showStartingFromMessage ? i18n('SHARED__EARLY_BIRD__PER_PASSENGER_PRICE_MESSAGE_PREFIX') : '(';
};

const getEarlyBirdUnitAmount = (unitPriceInBound: CurrencyType, unitPriceOutBound: CurrencyType) => {
  const earlyBirdUnitPrice = _.chain([unitPriceOutBound, unitPriceInBound])
    .compact()
    .minBy((item) => numeral(item.amount).value())
    .value();

  return _.get(earlyBirdUnitPrice, 'amount');
};

export const getEarlyBirdAdditionalTemplateData = (
  unitPriceInBound: CurrencyType,
  unitPriceOutBound: CurrencyType,
  earlyBirdEligibility: ?EarlyBirdEligibility
) => ({
  earlyBirdTotalPrice: _.get(earlyBirdEligibility, 'totalPrice.amount'),
  earlyBirdUnitPrice: getEarlyBirdUnitAmount(unitPriceInBound, unitPriceOutBound),
  earlyBirdPrefixText: getEarlyBirdPrefixText(unitPriceInBound, unitPriceOutBound)
});
