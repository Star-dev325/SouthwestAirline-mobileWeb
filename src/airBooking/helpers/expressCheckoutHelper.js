// @flow
import _ from 'lodash';
import type { Passenger } from 'src/shared/flow-typed/shared.types';
import validator from 'src/shared/form/formValidators/validator';

export const isInformationCompletedForExpressCheckout = (passengerDetailsPage: Passenger): boolean => {
  const { firstName, lastName, dateOfBirth, gender } = passengerDetailsPage || {};

  return !_.isEmpty(firstName) && !_.isEmpty(lastName) && !_.isEmpty(dateOfBirth) && validator.isValidGender(gender);
};
