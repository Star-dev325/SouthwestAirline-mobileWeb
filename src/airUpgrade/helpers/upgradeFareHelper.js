// @flow
import type { UpgradeFareReservationDataType } from 'src/shared/flow-typed/shared.types';
import type { PricingDataType, BoundSelectionDataType } from 'src/airUpgrade/flow-typed/airUpgrade.types';

export const buildUpgradeFareReservationRequest = ({
  firstName,
  lastName,
  recordLocator,
  passengerSearchToken,
  nonPremiumSearch,
  link
}: UpgradeFareReservationDataType) =>
  (link?.body?.passengerSearchToken
    ? link
    : {
      href: `/v1/mobile-air-booking/page/upgrade/${recordLocator}`,
      body: {
        firstName,
        lastName,
        recordLocator,
        passengerSearchToken,
        nonPremiumSearch
      },
      method: 'POST'
    });

export const getPricingChangeRequests = (
  changePricingLink: Link,
  pricingDataList: Array<PricingDataType>,
  boundSelectionDataList: Array<BoundSelectionDataType>
) => {
  const changeRequests = changePricingLink.body?.changeRequests ?? [];
  const fundsAppliedToken = changePricingLink.body?.fundsAppliedToken;

  const changeRequestsWithProductId = boundSelectionDataList.map<BoundSelectionDataType>(
    (boundSelectionDataType, index): BoundSelectionDataType => {
      const upgradeChangeRequest = changeRequests[index];

      if (boundSelectionDataType.canUpgrade) {
        const isSelected = pricingDataList.some(
          (pricingData) => pricingData.productId === boundSelectionDataType.productId && pricingData.isSelected
        );

        return isSelected
          ? {
            ...upgradeChangeRequest,
            productId: boundSelectionDataType.productId,
            arrivalAirportCode: boundSelectionDataType.arrivalAirportCode,
            departureAirportCode: boundSelectionDataType.departureAirportCode
          }
          : { ...upgradeChangeRequest };
      }

      return { ...upgradeChangeRequest };
    }
  );

  return { changeRequests: changeRequestsWithProductId, fundsAppliedToken };
};
