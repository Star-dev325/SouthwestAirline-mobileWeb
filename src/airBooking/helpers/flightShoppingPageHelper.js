import _ from 'lodash';
import { permutations } from 'src/shared/helpers/mathUtils';
import { INBOUND, OUTBOUND } from 'src/shared/constants/flightBoundTypes';
import PassengerTypes from 'src/shared/constants/passengerTypes';

const { ADULT } = PassengerTypes;

export const getNextFlightShoppingPageParams = (params, response) => {
  const { inboundPage } = response;
  const bounds = inboundPage ? [OUTBOUND, INBOUND] : [OUTBOUND];
  const paxTypes = [ADULT];

  const pages = permutations(paxTypes, bounds).map((paxAndBound) => ({
    paxType: paxAndBound[0],
    direction: paxAndBound[1]
  }));

  const index = _.findIndex(pages, params);

  return index === -1 ? undefined : _.get(pages, `[${index + 1}]`);
};

export const getFirstShoppingPageParams = () => ({
  paxType: ADULT,
  direction: OUTBOUND
});

export const generateFlightShoppingPages = (response) => {
  const { outboundPage, inboundPage } = response.flightShoppingPage;
  const isRoundTrip = !!inboundPage;
  let pages = [];

  const buildPages = () => {
    const result = [
      {
        direction: OUTBOUND,
        paxType: ADULT,
        ...outboundPage
      }
    ];

    if (isRoundTrip) {
      result.push({
        direction: INBOUND,
        paxType: ADULT,
        ...inboundPage
      });
    }

    return result;
  };

  pages = pages.concat(buildPages());

  return pages;
};
