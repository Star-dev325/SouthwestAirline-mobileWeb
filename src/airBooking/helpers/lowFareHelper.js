// @flow
import _ from 'lodash';
import dayjs from 'dayjs';
import type {
  LowFareCalendarDaysType,
  LowFareByMonthArraysType,
  LowFareCalendarPageType
} from 'src/airBooking/flow-typed/lowFare.types';
import { convertToNumber } from 'src/shared/helpers/numberHelper';
import { isPastDate } from 'src/shared/helpers/dateHelper';

export const MIN_BAR_HEIGHT_WITH_TAXES = 4.5;
export const MIN_BAR_HEIGHT = 3.7;
export const MAX_BAR_HEIGHT = 19;

const SMOOTH_SCROLL_TIME = 300;
const REM_CONVERSION = 10;

export const hasPricePoints = (boundFlights: Array<LowFareCalendarDaysType>) =>
  !_.chain(boundFlights)
    .map((fare) => !_.isEmpty(_.get(fare, 'lowestPrice.pricePointsTax')))
    .uniq()
    .compact()
    .isEmpty()
    .value();

export const getMinPrice = (boundFlights: Array<LowFareCalendarDaysType>, todayAsDate: string) =>
  _.chain(boundFlights)
    .map((fare) => {
      const value = _.get(fare, 'lowestPrice.price.amount', undefined);
      const amount = convertToNumber(value);
      const min = isPastDate(fare.date, todayAsDate) ? undefined : amount;

      return min;
    })
    .min()
    .value() || 0;

export const getMaxPrice = (boundFlights: Array<LowFareCalendarDaysType>, todayAsDate: string) =>
  _.chain(boundFlights)
    .map((fare) => {
      const value = _.get(fare, 'lowestPrice.price.amount', 0);
      const amount = convertToNumber(value);

      return isPastDate(fare.date, todayAsDate) ? 0 : amount;
    })
    .max()
    .value() || 0;

const percentageInPriceRange = (farePrice: number, minPrice: number, maxPrice: number): number => {
  const denom: number = maxPrice - minPrice;

  if (denom === 0) {
    return 1;
  }

  return (farePrice - minPrice) / denom;
};

export const getBarHeight = (
  farePrice: ?number,
  minPrice: number,
  maxPrice: number,
  showPointsTax: boolean
): string => {
  const minBarHeight = showPointsTax ? MIN_BAR_HEIGHT_WITH_TAXES : MIN_BAR_HEIGHT;

  if (!maxPrice || !minPrice || farePrice === 0 || farePrice === undefined || farePrice === null) {
    return `${minBarHeight}rem`;
  }

  const percentage = percentageInPriceRange(farePrice, minPrice, maxPrice);
  const adjustedHeight = MAX_BAR_HEIGHT * percentage + minBarHeight * (1 - percentage);

  return `${adjustedHeight}rem`;
};

export const getLowFareDaysByMonth = (boundFares: Array<LowFareCalendarDaysType>): LowFareByMonthArraysType => {
  const lowFareDaysMonthMap = new Map();
  let faresInThisMonth: Array<LowFareCalendarDaysType> = [];
  let currentMonthYear: string; // 1 = Jan ... 12 = Dec  Ex. 1-2019

  _.forEach(boundFares, (fare: LowFareCalendarDaysType) => {
    const thisDate = dayjs(fare.date);

    currentMonthYear = thisDate.format('M-YYYY');

    faresInThisMonth = lowFareDaysMonthMap.get(currentMonthYear) || [];
    faresInThisMonth.push(fare);
    lowFareDaysMonthMap.set(currentMonthYear, faresInThisMonth);
  });

  const lowFareDaysMonthArray: LowFareByMonthArraysType = [];

  for (const value of lowFareDaysMonthMap.values()) {
    lowFareDaysMonthArray.push(value);
  }

  return lowFareDaysMonthArray;
};

export const calculateLfcSelectedDates = (searchRequest: *, response: LowFareCalendarPageType) => {
  const { departureDate, returnDate } = searchRequest;
  const outboundDays = _.get(response, 'lowFareCalendarPage.outboundPage.lowFareCalendarDays');
  const inboundDays = _.get(response, 'lowFareCalendarPage.inboundPage.lowFareCalendarDays');
  let selectedDepartureDate;
  let selectedReturnDate;

  if (
    departureDate &&
    outboundDays &&
    outboundDays.filter((d) => d.date === departureDate && d.lowestPrice).length > 0
  ) {
    selectedDepartureDate = departureDate;
  }

  if (returnDate && inboundDays && inboundDays.filter((d) => d.date === returnDate && d.lowestPrice).length > 0) {
    selectedReturnDate = returnDate;
  }

  return { selectedDepartureDate, selectedReturnDate };
};

export const calculateCalendarScrollableBounds = (calendar: *, leftSpacer: *, rightSpacer: *) => {
  const calDays = calendar.getElementsByClassName('calendar-day');
  const firstBar = calDays[0];
  const lastBar = calDays[calDays.length - 1];
  const outboundCalRefRect = calendar.getBoundingClientRect();
  const firstBarRect = firstBar.getBoundingClientRect();
  const lastBarRect = lastBar.getBoundingClientRect();
  const calMidPoint = outboundCalRefRect.left + outboundCalRefRect.width / 2;
  const diffLeftToCenter = Math.round(calMidPoint - (firstBar.offsetLeft + firstBarRect.width / 2));
  const diffRightToCenter = Math.round(
    calMidPoint - (calendar.scrollWidth - (lastBar.offsetLeft + lastBarRect.width / 2))
  );
  const calendarPaddingLeft = diffLeftToCenter / REM_CONVERSION;
  const calendarPaddingRight = diffRightToCenter / REM_CONVERSION;

  if (calendarPaddingLeft) {
    const currentPaddingLeft = leftSpacer.style['paddingLeft']
      ? parseFloat(leftSpacer.style['paddingLeft'].replace('rem', ''))
      : 0;

    leftSpacer.style['paddingLeft'] = `${currentPaddingLeft + calendarPaddingLeft}rem`;
  }

  if (calendarPaddingRight) {
    const currentPaddingRight = rightSpacer.style['paddingRight']
      ? parseFloat(rightSpacer.style['paddingRight'].replace('rem', ''))
      : 0;

    rightSpacer.style['paddingRight'] = `${currentPaddingRight + calendarPaddingRight}rem`;
  }
};

export const calendarScrollFunctions = {
  focusOnPriceBar: (lowFareCalendar: *, targetEl: HTMLElement, noAnimation?: boolean) => {
    if (lowFareCalendar) {
      const lfcClientRect = lowFareCalendar.getBoundingClientRect();
      const targetClientRect = targetEl.getBoundingClientRect();
      const differenceFromCenter = Math.round(
        targetClientRect.left + targetClientRect.width / 2 - (lfcClientRect.left + lfcClientRect.width / 2)
      );
      const positionToScrollTo = lowFareCalendar.scrollLeft + differenceFromCenter;
      const startTime = _.has(window, 'performance.now') ? performance.now() : new Date().getTime();

      calendarScrollFunctions.scrollToPriceBar(
        noAnimation,
        lowFareCalendar,
        positionToScrollTo,
        startTime,
        lowFareCalendar.scrollLeft
      );
    }
  },
  scrollToPriceBar: (
    skipAnimation: boolean = false,
    elementToScroll: HTMLElement = document.createElement('span'),
    positionToScrollTo: number = 0,
    startTime: number = 0,
    startPosition: number = 0
  ) => {
    if (skipAnimation) {
      elementToScroll.scrollLeft = positionToScrollTo;

      return;
    }

    const currentTime = _.has(window, 'performance.now') ? performance.now() : new Date().getTime();
    const progress = (currentTime - startTime) / SMOOTH_SCROLL_TIME;
    const percentageComplete = Math.min(1, progress);
    const newPosition = Math.ceil(
      easeOutQuad(percentageComplete) * (positionToScrollTo - startPosition) + startPosition
    );

    if (elementToScroll.scrollLeft === positionToScrollTo || isAtEndOfScroll(elementToScroll, percentageComplete)) {
      return; // end scrolling
    }

    elementToScroll.scrollLeft = newPosition;

    if (_.has(window, 'requestAnimationFrame')) {
      window.requestAnimationFrame(
        calendarScrollFunctions.scrollToPriceBar.bind(
          null,
          false,
          elementToScroll,
          positionToScrollTo,
          startTime,
          startPosition
        )
      );
    }
  }
};

const easeOutQuad = (t: number) => t * (2 - t);

const isAtEndOfScroll = (elementToScroll: HTMLElement, percentageComplete: number) => {
  if (percentageComplete !== 1) {
    return false;
  }

  return elementToScroll.scrollLeft === 0 ||
    elementToScroll.offsetWidth + elementToScroll.scrollLeft === elementToScroll.scrollWidth ||
    percentageComplete >= 1;
};
