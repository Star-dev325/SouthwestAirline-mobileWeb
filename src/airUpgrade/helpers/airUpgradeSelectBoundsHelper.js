import { addCurrency } from 'src/shared/api/helpers/currencyHelper';
import { toNumberFromFormattedString, toFormattedStringFromNumber } from 'src/shared/helpers/currencyValueHelper';
import { POINTS } from 'src/shared/constants/currencyTypes';

export const flattenUpgradeBounds = (upgradeBounds) =>
  upgradeBounds.reduce((flattenedBounds, upgradeBound) => {
    const { boundSelectionDataList = [], upgradeablePricingDataList = [] } = flattenedBounds;

    const {
      arrivalAirportCode,
      arrivalTime,
      boundType,
      departureAirportCode,
      departureDate,
      departureDayOfWeek,
      departureTime,
      flight,
      isOvernight,
      isNextDayArrival,
      numberOfPassengers,
      upgradeFare: {
        upgradePrice = null,
        upgradeTotalPrice = null,
        upgradeMessage: { header: upgradeMessageHeader, body: upgradeMessageBody },
        _meta: { productId, canUpgrade }
      }
    } = upgradeBound;

    const boundSelectionData = {
      arrivalAirportCode,
      arrivalTime,
      boundType,
      canUpgrade,
      departureAirportCode,
      departureDate,
      departureDayOfWeek,
      departureTime,
      isOvernight,
      isNextDayArrival,
      productId,
      upgradeMessageBody,
      upgradeMessageHeader
    };

    const pricingData = {
      arrivalAirportCode,
      boundType,
      canUpgrade,
      departureAirportCode,
      flight,
      isSelected: false,
      numberOfPassengers,
      productId,
      upgradePrice,
      upgradeTotalPrice
    };

    return {
      boundSelectionDataList: [...boundSelectionDataList, boundSelectionData],
      upgradeablePricingDataList: canUpgrade ? [...upgradeablePricingDataList, pricingData] : upgradeablePricingDataList
    };
  }, {});

export const updateSelectedPricingData = (pricingDataList, { productId, isSelected }) =>
  pricingDataList.map((pricingData) =>
    (pricingData.productId === productId ? { ...pricingData, isSelected } : pricingData)
  );

export const addPoints = (...arrayOfCurrencyType) => {
  const { currencyCode = POINTS, currencySymbol = '', amount: unformattedAmount } = addCurrency(...arrayOfCurrencyType);
  const amountAsNumber = toNumberFromFormattedString(unformattedAmount);
  const amount = toFormattedStringFromNumber(amountAsNumber);

  return {
    currencyCode,
    currencySymbol,
    amount
  };
};
