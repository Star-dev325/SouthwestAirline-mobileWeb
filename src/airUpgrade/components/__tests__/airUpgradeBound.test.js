import { fireEvent, render, waitFor } from '@testing-library/react';
import React from 'react';
import AirUpgradeBound from 'src/airUpgrade/components/airUpgradeBound';
import AirUpgradeViewReservationApiJsonBuilder from 'test/builders/apiResponse/airUpgradeViewReservationApiJsonBuilder';
import { createMockedForm, createMockedFormStore } from 'test/unit/helpers/formTestUtils';

describe('AirUpgradeBound', () => {
  let onSubmitStub;

  beforeEach(() => {
    onSubmitStub = jest.fn();
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  const getBoundSelectionData = () =>
    new AirUpgradeViewReservationApiJsonBuilder().build().viewUpgradeReservationPage.boundSelectionDataList[0];

  it('Should render a selectedable bound if it can be upgraded', () => {
    const boundData = getBoundSelectionData();
    const { asFragment } = createComponent(boundData);

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render a message explaining why the bound is not selectable if it is not upgradeable', () => {
    const boundData = {
      ...getBoundSelectionData(),
      canUpgrade: false,
      upgradeMessageHeader: 'mock cannot upgrade reason'
    };
    const { asFragment } = createComponent(boundData);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call the on change handler when the checkbox state changes', async () => {
    const boundData = getBoundSelectionData();
    const onChange = jest.fn();
    const { container } = createComponent(boundData, onChange);

    const checkbox = container.querySelector('.field .checkbox-button');

    fireEvent.click(checkbox);

    await waitFor(() => {
      expect(onChange).toHaveBeenCalledWith({
        isSelected: true,
        productId: boundData.productId
      });
    });
  });

  describe('overnight', () => {
    it('should show overnight indicator when isOvernight is true', () => {
      const boundData = {
        ...getBoundSelectionData(),
        isOvernight: true
      };

      const { container } = createComponent(boundData);

      expect(container.querySelector('[data-qa="overnight-indicator"]')).toMatchSnapshot();
    });

    it('should not show overnight indicator when isOvernight is false', () => {
      const boundData = {
        ...getBoundSelectionData(),
        isOvernight: false
      };

      const { container } = createComponent(boundData);

      expect(container.querySelector('[data-qa="overnight-indicator"]')).toMatchSnapshot();
    });
  });

  describe('isNextDay', () => {
    it('should show next day indicator when isNextDayArrival is true', () => {
      const boundData = {
        ...getBoundSelectionData(),
        isNextDayArrival: true
      };

      const { container } = createComponent(boundData);

      expect(container.querySelector('[data-qa="next-day-indicator"]')).toMatchSnapshot();
    });

    it('should not show next day indicator when isNextDayArrival is false', () => {
      const boundData = {
        ...getBoundSelectionData(),
        isNextDayArrival: false
      };

      const { container } = createComponent(boundData);

      expect(container.querySelector('[data-qa="next-day-indicator"]')).toMatchSnapshot();
    });
  });

  const createComponent = (boundData, onChange = () => {}) => {
    const store = createMockedFormStore();
    const MockedForm = createMockedForm(store, { clickableChildren: true });

    return render(
      <MockedForm formData={{ formCheckBox: true }} onSubmit={onSubmitStub}>
        <AirUpgradeBound boundData={boundData} onChange={onChange} />
      </MockedForm>
    );
  };
});
