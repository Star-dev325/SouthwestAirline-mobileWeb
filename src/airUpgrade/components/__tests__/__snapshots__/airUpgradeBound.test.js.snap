// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirUpgradeBound Should render a message explaining why the bound is not selectable if it is not upgradeable 1`] = `
<DocumentFragment>
  <form>
    <div
      class="air-upgrade-bound"
    >
      <div
        class="air-upgrade-bound--type"
      >
        DEPARTING
      </div>
      <div
        class="flight-brief-summary bgwhite bdb p5"
      >
        <div
          class="flex xlarge pdkblue bold"
        >
          <div
            class="flex4"
          >
            Dec 29
          </div>
          <div
            class="flex2"
          >
            DAL
          </div>
          <div
            class="flex2 flight-brief-summary--airplane center"
          >
            <i
              class="mt3 icon icon_airplane"
            />
          </div>
          <div
            class="flex2"
          >
            HOU
          </div>
        </div>
        <div
          class="flex large gray4"
        >
          <div
            class="flex4"
          >
            Wednesday
          </div>
          <div
            class="flex2 flight-time"
          >
            <div
              class="flight-time--time"
            >
              <span>
                9:10
              </span>
              <span
                class="time-period"
              >
                AM
              </span>
            </div>
          </div>
          <div
            class="flex2"
          />
          <div
            class="flight-brief-summary--summary-group"
          >
            <div
              class="flex2 flight-time"
            >
              <div
                class="flight-time--time"
              >
                <span>
                  3:40
                </span>
                <span
                  class="time-period"
                >
                  PM
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="air-upgrade-bound--no-upgrade-msg"
      >
        mock cannot upgrade reason
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`AirUpgradeBound Should render a selectedable bound if it can be upgraded 1`] = `
<DocumentFragment>
  <form>
    <div
      class="air-upgrade-bound"
    >
      <div
        class="air-upgrade-bound--type"
      >
        DEPARTING
      </div>
      <div
        class="flight-brief-summary bgwhite bdb p5"
      >
        <div
          class="flex xlarge pdkblue bold"
        >
          <div
            class="flex4"
          >
            Dec 29
          </div>
          <div
            class="flex2"
          >
            DAL
          </div>
          <div
            class="flex2 flight-brief-summary--airplane center"
          >
            <i
              class="mt3 icon icon_airplane"
            />
          </div>
          <div
            class="flex2"
          >
            HOU
          </div>
        </div>
        <div
          class="flex large gray4"
        >
          <div
            class="flex4"
          >
            Wednesday
          </div>
          <div
            class="flex2 flight-time"
          >
            <div
              class="flight-time--time"
            >
              <span>
                9:10
              </span>
              <span
                class="time-period"
              >
                AM
              </span>
            </div>
          </div>
          <div
            class="flex2"
          />
          <div
            class="flight-brief-summary--summary-group"
          >
            <div
              class="flex2 flight-time"
            >
              <div
                class="flight-time--time"
              >
                <span>
                  3:40
                </span>
                <span
                  class="time-period"
                >
                  PM
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="air-upgrade-bound--select"
      >
        <div
          class="field"
        >
          <div
            class="relative flex checkbox-button flex-cross-center"
            name="mockId1"
          >
            <div>
              <span
                class="checkbox-button--mark"
              >
                <i
                  class="absolute xlarge bgtransp icon icon_check"
                />
              </span>
            </div>
            <div>
              <div
                class="checkbox-button--children flex-equal-width ml2"
              >
                <div
                  class="air-upgrade-bound--select-message-header"
                >
                  Upgrade all passengers to Business SelectÂ®
                </div>
                <div
                  class="air-upgrade-bound--select-message-body"
                >
                  and earn 6,937 more points
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`AirUpgradeBound isNextDay should not show next day indicator when isNextDayArrival is false 1`] = `null`;

exports[`AirUpgradeBound isNextDay should show next day indicator when isNextDayArrival is true 1`] = `
<div
  class="multi-day-indicator multi-day-indicator--nextday-green"
  data-qa="next-day-indicator"
>
  <i
    class="multi-day-indicator--icon multi-day-indicator--icon-small icon icon_next-day-indicator"
  />
  <p
    class="multi-day-indicator--font-small"
  >
    AIR_BOOKING__SHOPPING__NEXT_DAY
  </p>
</div>
`;

exports[`AirUpgradeBound overnight should not show overnight indicator when isOvernight is false 1`] = `null`;

exports[`AirUpgradeBound overnight should show overnight indicator when isOvernight is true 1`] = `
<div
  class="multi-day-indicator multi-day-indicator--overnight-red"
  data-qa="overnight-indicator"
>
  <i
    class="multi-day-indicator--icon multi-day-indicator--icon-small icon icon_next-day-indicator"
  />
  <p
    class="multi-day-indicator--font-small"
  >
    AIR_BOOKING__SHOPPING__OVERNIGHT
  </p>
</div>
`;
